src/main/java/com/guciowons/shoppingbasket/Product/ProductService.java

```diff
@@ -30,13 +33,19 @@ public void refreshProductsInDatabase(){
     public void insertProducts(){
             productProvider.getProducts()
                     .forEach(externalProduct -> productRepository.findProductByExternalId(externalProduct.getExternalId()).ifPresentOrElse(
-                            databaseProduct -> updateProducts(databaseProduct, externalProduct),
-                            () -> productRepository.save(externalProduct)
+                            databaseProduct -> updateProduct(databaseProduct, externalProduct),
+                            () -> insertProduct(externalProduct)
                     ));
     }
 
-    private void updateProducts(Product databaseProduct, Product externalProduct){
+    private void insertProduct(Product externalProduct){
```

Comment:
Teraz jak tutaj robisz podwójnego inserta dla każdego rekordu to co się stanie jak zapis produktu przejdzie dobrze, a zapis historii cen się wywali? Dane będą niespójne -- poczytaj sobie jak zrobić transakcje bazodanowe w springu.